HousingSearchListener/Infrastructure/BaseFunction.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using Amazon.XRay.Recorder.Handlers.AwsSdk;
+using Elasticsearch.Net;
+using Hackney.Core.Logging;
+using HousingSearchListener.Gateways;
+using HousingSearchListener.V1.HealthCheck;
+using HousingSearchListener.V1.Interfaces;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Nest;
+using HttpClientHandler = HousingSearchListener.V1.Interfaces.HttpClientHandler;
+
+namespace HousingSearchListener.Infrastructure
+{
+    [ExcludeFromCodeCoverage]
+    public abstract class BaseFunction
+    {
+        protected IConfigurationRoot Configuration { get; }
+
+        protected IServiceProvider ServiceProvider { get; }
+
+        protected ILogger Logger { get; }
+
+        protected BaseFunction(IServiceCollection services)
+        {
+            AWSSDKHandler.RegisterXRayForAllServices();
+
+            var builder = new ConfigurationBuilder();
+
+            Configuration = builder.Build();
+            RegisterDependencies(services);
```

Comment:
I would have put this call in the derived class `ConfigureServices()` implementation.
This BaseFunction class is meant to be a generic thing and derived instances would override `ConfigureServices()` to add just the stuff they needed.